src/d3a/models/strategy/__init__.py

```diff
@@ -32,10 +32,25 @@
 from d3a.d3a_core.util import append_or_create_key
 from d3a.models.market.market_structures import trade_from_JSON_string, offer_from_JSON_string
 from d3a.d3a_core.redis_connections.redis_area_market_communicator import BlockingCommunicator
+from d3a.constants import FLOATING_POINT_TOLERANCE
 
 log = getLogger(__name__)
 
 
+def assert_if_trade_bid_price_is_too_high(strategy, market, trade):
```

Comment:
I think you can make this function and the next as methods of the BaseStrategy class. That way you do not have to pass the self object as a first argument.

Reply:
You are right! Will do it.